fix: enhance/bug-fix skills visibility handling for older clients

libergod opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Behavior

### **Actual**

In older versions, the newWindowButton is hidden and occupies the space. Needed to change anchors from lockButton to be realigned at the side of the ContextMenuButton.
<img width="176" height="404" alt="image" src="https://github.com/user-attachments/assets/df3b2877-db14-4782-a3b3-833d58bdcfd7" />

### **Expected**

<img width="281" height="718" alt="image" src="https://github.com/user-attachments/assets/b861c8be-a12f-4a03-a0eb-73f6c3d728b2" />


Do this and that happens

## Fixes

\# (issue)

## Type of change

Please delete options that are not relevant.

  - [X] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

  - [ ] Test A
  - [ ] Test B

**Test Configuration**:

  - Server Version: CANARY 1412
  - Client: OTC 
  - Operating System: W11

## Checklist

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my own code
  - [X] I checked the PR checks reports
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [X] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings
  - [X] I have added tests that prove my fix is effective or that my feature works

libergod committed:
improve: enhance skills visibility handling for older clients

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -78,6 +78,24 @@ function init()
     local newWindowButton = skillsWindow:recursiveGetChildById('newWindowButton')
+    
+    -- Check client version to handle newWindowButton visibility and lockButton positioning
+    if g_game.getClientVersion() < 1310 and newWindowButton then
```
If you put `g_game.getClientVersion() `in `onInit()`, it will always be 0

https://github.com/mehah/otclient/blob/8311a011a0c5730eb8501332db79013b8d5b9d3c/src/client/game.h#L1028

libergod replied:
I disagree. Made a couple of checks, and if correctly set up on the init.lua, it will run perfectly. 

<img width="2222" height="1052" alt="image" src="https://github.com/user-attachments/assets/7439956a-011f-4188-8218-acb812255d22" />

Anyway, will remove it and maintain only the online() function.

kokekanon replied:
@libergod 
When you run onInit(), there is no version set, so it will always return 0. The only way to get the version during onInit() is by having this configured init.lua 
https://github.com/mehah/otclient/blob/8311a011a0c5730eb8501332db79013b8d5b9d3c/init.lua#L13-L23
or
 by reload.

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -725,8 +747,20 @@ function online()
 
-    if g_game.getClientVersion() < 1310 then
-        newWindowButton:hide()
+    if g_game.getClientVersion() < 1310 and newWindowButton then
```
Why is this code repeated x3 ? L83 vs L750 vs L1071

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -1029,6 +1066,24 @@ function toggle()
         local newWindowButton = skillsWindow:recursiveGetChildById('newWindowButton')
+        
+        -- Check client version to handle newWindowButton visibility and lockButton positioning
+        if g_game.getClientVersion() < 1310 and newWindowButton then
```
Why is this code repeated x3 ? L83 vs L750 vs L1071

libergod replied:
Will remove it and maintain only the online() function. Thanks

kokekanon commented:
feedback

# 1) 
There is unnecessary code, just to hide a double horizontal separator

<img width="260" height="68" alt="image" src="https://github.com/user-attachments/assets/91e57985-30c4-4497-bb69-f4a9036ff60a" />

 . Just hide the second separator at gameStart.
example in L777
```lua
 if g_game.getFeature(GameEnterGameShowAppearance) then -- Tibia Qt
     skillsWindow:recursiveGetChildById('regenerationTime'):getChildByIndex(1):setText('Food') 
 else -- old client
    2ndHorizontalSeparator:hide() 
 end 
```
# 2) 

> Why is this code repeated x3 ? L83 vs L750 vs L1071

try moving this block of code  

> -- Check client version to handle newWindowButton visibility and lockButton positioning 
> if g_game.getClientVersion() < 1310 and newWindowButton then


to L776

https://github.com/mehah/otclient/blob/066850af349f385e96e25ce94f4fa87faa6572ff/modules/game_skills/skills.lua#L775-L777


# 3) 

also add this:

```lua
skillsWindow:recursiveGetChildById('experience'):getChildByIndex(1):setText('XP')
```
(correct label for tibia +13)[I had done it, but Nottinghster changed it.]
example 
```diff
 if g_game.getFeature(GameEnterGameShowAppearance) then 
     skillsWindow:recursiveGetChildById('regenerationTime'):getChildByIndex(1):setText('Food') 
+    skillsWindow:recursiveGetChildById('experience'):getChildByIndex(1):setText('XP')
+--  behavior of lockButton and contextMenuButton
 end 
```

libergod committed:
improve: adjust skill visibility handling for older client versions

libergod committed:
Improve: Code Refactor, Clean and Improve.

Was missing values from skills, as well as the save/load when checking/unchecking the Offence, Defence and Misc Stats.

libergod committed:
Update skills.lua

libergod committed:
final fix

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -699,48 +734,43 @@ function update()
+        skillsWindow:recursiveGetChildById('experience'):getChildByIndex(1):setText('XP')
+        
+        local newWindowButton = skillsWindow:recursiveGetChildById('newWindowButton')
+        if g_game.getClientVersion() < 1310 and newWindowButton then
```
8.6
<img width="181" height="97" alt="image" src="https://github.com/user-attachments/assets/2221bbb3-e816-4ff7-93a6-4328ff8ba7f8" />

This doesn't make sense.
The `GameEnterGameShowAppearance `feature applies to version 12.00 but when making another "if condition "inside < 13.10.

This entire block of code would apply between versions 12.00 and 13.09, but your goal was for it not to be visible in older versions, right?

You should have added it inside an else.

```lua
   if g_game.getFeature(GameEnterGameShowAppearance) then -- +12.00 (Tibia Qt)
        skillsWindow:recursiveGetChildById('regenerationTime'):getChildByIndex(1):setText('Food')
        skillsWindow:recursiveGetChildById('experience'):getChildByIndex(1):setText('XP')
        
        else -- (< 12.00) [10.98, 8.6 , 7.72, etc..]
hide newWindowButton and reorder lockButton 
end
```

kokekanon commented:
I don't understand something. The purpose of this PR was only to hide a horizontalSeparator and reorder a button in the miniwindows, for old protocol
but
<img width="106" height="37" alt="image" src="https://github.com/user-attachments/assets/ac28d932-0cb9-472a-8901-ffdcd9acd9f9" /> :skull:

libergod commented:
> I don't understand something. The purpose of this PR was only to hide a horizontalSeparator and reorder a button in the miniwindows, for old protocol but <img alt="image" width="106" height="37" src="https://private-user-images.githubusercontent.com/114332266/483738929-ac28d932-0cb9-472a-8901-ffdcd9acd9f9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0ODkyMzYsIm5iZiI6MTc1NjQ4ODkzNiwicGF0aCI6Ii8xMTQzMzIyNjYvNDgzNzM4OTI5LWFjMjhkOTMyLTBjYjktNDcyYS04OTAxLWZmZGNkOWFjZDlmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyOVQxNzM1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NmJkOGJlYWM0MTVjNGIyNzQ2N2M5M2M1NTY5ZjNmM2VkYmRmMWVkNjZmNmRhZTI3MmE0NWU4ZTM5ZGNjY2JmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pUxZxzS1opMdCx7EjHtHoU6WK3PJjvLUZhVtbe0nLF8"> 💀

Improved the code. It was too messed up. I realized I was calling two, three times the same block of code. Now I cleaned it up and created functions to be called instead of repeating myself.

Also, noticed that some elemental skills were not being shown properly when in 1412, their values were not being set up

<img width="260" height="1292" alt="image" src="https://github.com/user-attachments/assets/5b84c33e-29a4-4aea-b8cb-211aab9e6810" />

kokekanon commented:
I believe that the Cipsoft client  14.12 does not display values of 0%. by default

libergod commented:
> I believe that the Cipsoft client 14.12 does not display values of 0%. by default

True, the correct behaviour is to hide it until values change. However, a more complex handling will be needed from the lua file, which I chose to do after. Just trying to fix the miniWindowButton and skills showing up to the correct client versions.

About the previous comments, I will take a closer look at it, might have misunderstood the checks about versions.

libergod committed:
improve: enhance skill visibility for older clients
